### ISSUE TITLE
Docker Hub builds are not being automatically build anymore

### ISSUE THREAD

**Issue by jdubois** [jhipster/generator-jhipster#6853]
This was done by @pascalgrimaud -> Pascal, if you have the time, I would appreciate your input here

Following http://www.jhipster.tech/docker-hub/ we should have automatic Docker builds of our sample applications.

As can been seen from our sample application at https://hub.docker.com/r/jhipster/jhipster-sample-app/tags/ , it hasn't been built for 9 month, and it is completely outdated.

We should find what went wrong, and correct his.

**Comment by ruddell** [comment]
The issue from Docker Hub is `Build failed: Dockerfile not found at ./Dockerfile`.  The last working version was v4.10.0 which has a [Dockerfile at the base](https://github.com/jhipster/jhipster-sample-app/blob/v4.1.0/Dockerfile)

The solution is to restore the Dockerfiles and add `-not name "Dockerfile"` to all of the `find` commands such as [this one](https://github.com/jhipster/jhipster-automated-samples/blob/master/generate_projects.sh#L219) in the sample app builder.  I can PR later tonight

**Comment by pascalgrimaud** [comment]
@jdubois : the Dockerfile at the root folder was probably deleted when you delete all files to regenerate something properly, in sample projects.

@ruddell : thanks for all PR but it's better to use this new Dockerfile, using the multi-stage builds:

```
FROM openjdk:8
ADD . /code/
RUN echo '{ "allow_root": true }' > /root/.bowerrc && \
    rm -Rf /code/target /code/node_modules && \
    cd /code/ && \
    ./mvnw clean package -Pprod -DskipTests && \
    mv /code/target/*.war /app.war

FROM openjdk:8-jre-alpine
ENV SPRING_OUTPUT_ANSI_ENABLED=ALWAYS \
    JHIPSTER_SLEEP=0 \
    JAVA_OPTS=""
EXPOSE 8080
CMD echo "The application will start in ${JHIPSTER_SLEEP}s..." && \
    sleep ${JHIPSTER_SLEEP} && \
    java ${JAVA_OPTS} -Djava.security.egd=file:/dev/./urandom -jar /app.war
COPY --from=0 /app.war .
```

**Comment by ruddell** [comment]
@pascalgrimaud No worries, I just cherry-picked your commits.  I updated the PRs to the new multi-stage Dockerfiles

**Comment by pascalgrimaud** [comment]
Thanks @ruddell you are not sleeping yet :)

**Comment by jdubois** [comment]
Yes thanks so much! Sorry to have messed up this!

**Comment by PierreBesson** [comment]
No this is quite understandable as this file is not generated by JHipster.

And this reminds me that I started a branch some time ago to add exactly this `root dockerfile` feature to the generator as a CI subgen option.
@jdubois is this something you would be interested in ?

**Comment by pascalgrimaud** [comment]
Right, I support it in my module, but as it doesn't require much maintenance and is easy to do, it would be cool to have it to our CICD subgen

**Comment by jdubois** [comment]
I need to have a look, what's the difference with the
src/main/docker/Dockerfile that we already generate?

Le 13 déc. 2017 6:35 AM, "Pascal Grimaud" <notifications@github.com> a
écrit :

**Comment by pascalgrimaud** [comment]
The Dockerfile inside `src/main/docker/Dockerfile` is used during the packaging process with: `./mvnw clean package -Pprod dockerfile:build`
That means: the machine needs to have Java installed and Docker

This new Dockerfile at root folder will only need Docker (Docker Hub don't have Java, just Docker), because the build process will be executed inside the 1st image which contains Java JDK, then it will copy the generated war file into a 2nd image which contains Java JRE.

I don't know if I'm clear enough
But personnaly, having it in my module or in CICD, is the same thing. It's just a suggestion from @PierreBesson

**Comment by jdubois** [comment]
OK @pascalgrimaud but I don't understand the point, outside our own use for DockerHub - what would "normal" people benefit from it?

**Comment by PierreBesson** [comment]
@jdubois This is for people to setup "automated builds", not necessarily with docker hub.

**Comment by jdubois** [comment]
Thanks @PierreBesson
